### PR TITLE
Milestone 4: Add Join Table for Visits

### PR DESCRIPTION
**Pull request by cherelemma**
# Summary of Implementation

In this fourth milestone, I wrote the following DD and DM queries. 

- Add the `vets`, `specializations`, and `visits` tables.
- Insert record into` vets`,  `specializations`, and `visits` tables.
- Write join table queries.

### Screen Shoots.
Find below the screen shoot of the query outputs.

- Who was the last animal seen by William Thatcher?

image

- How many different animals did Stephanie Mendez see?

image

- List all vets and their specialties, including vets with no specialties.

image

- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.

image

- What animal has the most visits to vets?

image

- Who was Maisy Smith's first visit?

image

- Details for most recent visit: animal information, vet information, and date of visit.

image

- How many visits were with a vet that did not specialize in that animal's species?

image

- What specialty should Maisy Smith consider getting? Look for the species she gets the most.

image

### CODE REVIEW

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: :

Hi @cherelemma , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :tophat:  :+1: 
- Good Summary :+1: 
- Good PR Description :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql [hunk 1]
```diff
@@ -25,3 +25,39 @@ UPDATE animals SET owner_id =
...
+INSERT INTO VETS (NAME, AGE, date_of_graduation) 
+VALUES ('William Tatcher', 45, '4/23/2000'), ('Maisy Smith', 26, '1/17/2019'), ('Stephanie Mendez', 64, '5/4/1981'), ('Jack Harkness', 38, '6/8/2008')
+
+INSERT INTO specializations (species_id, vets_id) 
+	SELECT s.id, v.id FROM vets v, species s JOIN animals a ON s.id = a.id WHERE s.name = 'Pokemon' AND v.name = 'William Tatcher'
+INSERT INTO specializations (species_id, vets_id) 
+	(SELECT s.id, v.id FROM vets v, species s JOIN animals a ON s.id = a.id WHERE s.name IN ('Digimon', 'Pokemon') AND v.name = 'Stephanie Mendez')
+INSERT INTO specializations (species_id, vets_id) 
+	SELECT s.id, v.id FROM vets v, species s JOIN animals a ON s.id = a.id WHERE s.name = 'Digimon' AND v.name = 'Jack Harkness'
+
```
Comment (GedeonTS) [inline data.sql:40]: - Kindly make Sure to add one more insertion to the `specialiazation` table that's missing, in you `data.sql`. There should be 4 insertions but you only have 3. Kindly add that.

File: queries.sql [hunk 2]
```diff
@@ -68,4 +68,34 @@ SELECT a.name FROM animals a
...
+	   GROUP BY animal_id ORDER BY num_visits DESC LIMIT 1) 
+	   AS visits_count ON a.id = visits_count.animal_id;
+
+SELECT a.name AS Animal_Name, vs.date_of_visit FROM visits vi
+	JOIN animals a ON vi.animal_id = a.id JOIN vets On vets.id = vi.vets_id
+	WHERE vets.name = 'Maisy Smith' ORDER BY vs.date_of_visit ASC LIMIT 1;
+
+SELECT a.name AS Animal_Name, a.date_of_birth, a.escape_attempts, a.neutered, a.weight_kg, 
+	ve.name AS Vet_Name, ve.age as Vet_Age, ve.date_of_graduation, vi.date_of_visit FROM animals a 
+	JOIN visits vi ON vi.animal_id = a.id JOIN vets ve On ve.id = vi.vets_id ORDER BY vi.date_of_visit DESC LIMIT 1 ;
```
Comment (GedeonTS) [inline queries.sql:99]: - Kindly make sure to add two more queries that are missing in your `query.sql` file which are :
       - How many times visits where with a vet that did not specialize in that animal's species?
       - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
       
Kindly add that too.

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: :

Hi @cherelemma , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :tophat:  :+1: 
- Good Summary :+1: 
- Good PR Description :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (cherelemma) [reply data.sql:40]: @GedeonTS  thank you for your constructive feedback. On this particular comment, have you seen this one? The second insertion statement adds two records at a time. I appreciate to if you consider it.
 
image

**Review by binhussen — APPROVED** [review]
Hi @cherelemma , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Well-Structured Query ✔️
- Descriptive pull request with screenshots ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**